src/client/Microsoft.Identity.Client/Internal/Broker/BrokerSilentRequest.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Internal.Broker
+{
+    internal class BrokerSilentRequest
+    {
+        internal Dictionary<string, string> BrokerPayload { get; set; } = new Dictionary<string, string>();
+        internal IBroker Broker { get; }
+        private readonly AcquireTokenSilentParameters _silentParameters;
+        private readonly AuthenticationRequestParameters _authenticationRequestParameters;
+        private readonly IServiceBundle _serviceBundle;
+        //private readonly AuthorizationResult _authorizationResult;
+
+        internal BrokerSilentRequest(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters acquireTokenSilentParameters,
+            IServiceBundle serviceBundle,
+            //AuthorizationResult authorizationResult,
+            IBroker broker)
+        {
+            _authenticationRequestParameters = authenticationRequestParameters;
+            _silentParameters = acquireTokenSilentParameters;
+            _serviceBundle = serviceBundle;
+            //_authorizationResult = authorizationResult;
+            Broker = broker;
+        }
+
+        public async Task<MsalTokenResponse> SendTokenRequestToBrokerAsync()
+        {
+            _authenticationRequestParameters.RequestContext.Logger.Info(LogMessages.CanInvokeBrokerAcquireTokenWithBroker);
+
+            return await SendAndVerifyResponseAsync().ConfigureAwait(false);
+        }
+
+        private async Task<MsalTokenResponse> SendAndVerifyResponseAsync()
+        {
+            CreateRequestParametersForBroker();
+
+            MsalTokenResponse msalTokenResponse =
+                await Broker.AcquireTokenUsingBrokerAsync(BrokerPayload).ConfigureAwait(false);
+
+            ValidateResponseFromBroker(msalTokenResponse);
+            return msalTokenResponse;
+        }
+
+        internal void CreateRequestParametersForBroker()
+        {
+            BrokerPayload.Clear();
```

Comment:
Why do we need to clear the payload? I think there is only one request object per request, these objects are not reused.